### DISCUSSION TITLE
Rendering the Asset[] props in any project interface could do with a working opensource example

### DISCUSSION THREAD

**Discussion by seantiz** [pieces-app/opensource#68] [Projects]
When I look at the docs page for the Asset endpoint it isn't clear which Asset property lets us load a code snippet into view.

I can destructure the Asset object and render props like `asset.name`  or `asset.id`, which is fun!

But that'll give the user a bunch of code snippet titles in view, without being able to actually click-through to read the snippet itself, like we can do in Pieces Desktop.

I'd be happy to help document this next step. But looking at the Asset object itself, I just don't know where that step is right now.

I've noticed that the Example Typescript playground looks like we hit pretty much the same wall there as I did in the Svelte x Pieces project on this topic.

**Comment by codeftsaikat** [comment]
I recommend checking repositories related to asset management or media galleries. Websites like GitHub or CodeSandbox often host real-world projects that can show how Asset[] props are handled.

Here’s how you can search for such examples:

GitHub: Search for "asset management React" or "media gallery React" in repositories. You'll find projects where assets like images, videos, or files are rendered dynamically from arrays.

CodeSandbox: Look for public templates of gallery applications or media players built with React.

These repositories often come with clear explanations on how assets are rendered using props like Asset[] and can give you hands-on examples for your project interface.

**Reply by seantiz** [reply]
Thanks a lot, this led me to "Digital Asset Management" pattern which does lay out a suggested pattern nicely.

I would personally still suggest the Assets API docs have:

- Some more inline comments for Pieces-idiomatic properties like `transferables` and `seed`
- A related link to an `Asset` example so that the page naturally flows to what we can do with Asset data props after we've received the response data
